### PR TITLE
feat(slack): Switch to workspace tokens

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This switches the slack integration to using Workspace tokens, which are tokens that are granted to an entire slack workspace, *not* just a single user.

This changes the linking flow, as Workspace tokens do not provide a way for users to authenticate with slack. Instead we now generate a signed link which includes the slack user ID who is intiating the linking from within slack.

### CODE REVIEW

File: src/sentry/integrations/slack/link_identity.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        req = session.post('https://slack.com/api/chat.postEphemeral', data=payload)
+        resp = req.json()
+        if not resp.get('ok'):
+            logger.error('slack.link-notify.response-error', extra={
+                'error': resp.get('error'),
+            })
+
+        messages.add_message(self.request, messages.SUCCESS, SLACK_IDENTITY_LINKED)
+
+        return HttpResponseRedirect('/')
```
Comment (dcramer) [inline src/sentry/integrations/slack/link_identity.py]: I'd prefer we respond with something a bit less janky at least before we launch to users.

(a basic "Your identity was linked. You can close this window.")

**Review by dcramer — APPROVED** [review]
can we also email existing slack beta users and let the mknow

File: src/sentry/integrations/slack/link_identity.py (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/integrations/slack/link_identity.py]: You thinking like just a simple single page with that message?

Comment (dcramer) [reply src/sentry/integrations/slack/link_identity.py]: yeah something like a traditional modal page (same kind of frame we use for login/etc)

Comment (EvanPurkhiser) [reply src/sentry/integrations/slack/link_identity.py]: 👍 can make that happen

**Comment by EvanPurkhiser** [comment]
image

**Comment by dcramer** [comment]
@EvanPurkhiser damn does go back to slack work

**Comment by dcramer** [comment]
also what if it was "Go back to Slack[ing off]"









jkjkjk

File: src/sentry/templates/sentry/slack-linked.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+{% load i18n %}
+
+{% block title %}{% trans "Slack Linked" %} | {{ block.super }}{% endblock %}
+{% block wrapperclass %}narrow auth{% endblock %}
+
+{% block main %}
+  <div class="align-center">
+    <p>
+      {% trans "Your slack identity has been associated with your Sentry account. You may now take actions through slack!" %}
```
Comment (mattrobenolt) [inline src/sentry/templates/sentry/slack-linked.html]: Don't you need to capitalize `Slack` here?

**Comment by EvanPurkhiser** [comment]
Yeah i do

**Comment by EvanPurkhiser** [comment]
> damn does go back to slack work

yes

**Pull request merged**